File: knotlink-sdk-cpp/tests/unit/nogui_opensocket_querier_test.cpp
// unit test: OpenSocketQuerier (nogui)
#include "../../knotlink/OpenSocketQuerier.hpp"
#include <iostream>
using namespace knotlink;

int main() {
    OpenSocketQuerier q;
    q.setConfig("app.knotlinksdk.test", "test_socket");
    try {
        std::string result = q.query_l("Hello, Responser!");
        std::cout << "Response: " << result << std::endl;
    } catch (const std::exception& e) {
        std::cerr << "Error: " << e.what() << std::endl;
    }
    return 0;
}

File: tests/cs-sdk/unit/nogui_opensocket_querier_test.cs
// unit test: OpenSocketQuerier (nogui)
using System;
using KnotLink;

var q = new OpenSocketQuerier("app.knotlinksdk.test", "test_socket");
try
{
    var result = q.Query("Hello, Responser!");
    Console.WriteLine($"Response: {result}");
}
catch (Exception e)
{
    Console.Error.WriteLine($"Error: {e.Message}");
}
q.Dispose();
